src/chunking.ts

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable no-bitwise, no-plusplus, no-mixed-operators */
+
+import type { Json } from '@metamask/utils';
+import { type TransportChunkFrame } from './types';
+
+const INT32_MAX = 0x7fffffff;
+
+let _randomCounter = 0 | 0;
+
+/**
+ * Creates a random ID for chunked messages. Always <= Int32 max value.
+ *
+ * @returns a random number that can be used as an ID
+ */
+function getNextId(): number {
+  // increment the counter and wrap around if it exceeds Int32 max value
+  _randomCounter = (_randomCounter + 1) & INT32_MAX;
+  return _randomCounter;
+}
+
+// half a "frame" (at 60 fps) per chunk, in milliseconds
+const FRAME_BUDGET_MS = 1000 / 60 / 2;
+
+/**
+ * Yield control to allow other tasks to run. Uses scheduler.yield() if available
+ * (modern browsers), otherwise falls back to setTimeout (older browsers).
+ */
+const tick = (() => {
+  return globalThis.scheduler?.yield
+    ? globalThis.scheduler.yield.bind(globalThis.scheduler)
```

Comment:
Is there a functional difference for our use case, will all browsers not support `setTimeout` for backwards compatibility?